File: classTypes4/c#/CarnivoroTerrestre.cs
// implementa la interfaz icarnivoro
public abstract class CarnivoroTerrestre : Mamifero, ICarnivoro 
{
    
    public CarnivoroTerrestre(string nombreCientifico, double pesoKg, int gestacionMeses) 
        // llama al constructor de la clase padre (mamifero)
        : base(nombreCientifico, pesoKg, gestacionMeses){
    }
    
    // implementacion del metodo de la interfaz icarnivoro
    public void cazar() {
        Console.WriteLine("el " + getNombreCientifico().ToLower() + " esta persiguiendo a su presa en la tierra");
    }
    
    // nuevo metodo abstracto para una tecnica especifica de caza.
    public abstract void tecnicadecaza();
}

File: classTypes4/java/src/CarnivoroTerrestre.java
// implementa la interfaz Carnivoro
public abstract class CarnivoroTerrestre extends Mamifero implements Carnivoro {
    
    public CarnivoroTerrestre(String nombreCientifico, double pesoKg, int gestacionMeses) {
        super(nombreCientifico, pesoKg, gestacionMeses);
    }
    
    // implementacion del método de la interfaz Carnivoro
    @Override
    public void cazar() {
        System.out.println("El " + getNombreCientifico() + " esta persiguiendo a su presa en la tierra");
    }
    
    // Nuevo metodo abstracto para una tecnica específica de caza.
    public abstract void tecnicaDeCaza();
}